src/targets/gpu/mlir.cpp

```diff
@@ -320,7 +320,8 @@ struct mlir_program
 
     MlirType make_tensor(const shape& s) const
     {
-        assert(s.standard());
+        assert(s.standard() && "MLIR expects all tensors to be in standard shape");
+        assert(not s.dynamic() && "MLIR does not support dynamic shapes");
```

Comment:
Use MIGRAPHX_THROW to be more consistent ? Assert would only work for debug  builds

Reply:
... yeah, good point, those should be throws